bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -0,0 +1,301 @@
+        updateAutomowerState();
+    }
+
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            config = getConfigAs(AutomowerConfiguration.class);
+
+            final String configMowerId = config.getMowerId();
+            if (configMowerId != null) {
+                automowerId.set(configMowerId);
+
+                // bridgeHandler.registerLightStatusListener(this);
+                startAutomowerPolling();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/conf-error-no-mower-id");
+            }
+
+        } else {
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+        }
+
+    }
+
+    @Nullable
+    private AutomowerBridge getAutomowerBridge() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            ThingHandler handler = bridge.getHandler();
+            if (handler instanceof AutomowerBridgeHandler) {
+                AutomowerBridgeHandler bridgeHandler = (AutomowerBridgeHandler) handler;
+                return bridgeHandler.getAutomowerBridge();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        if (!automowerId.get().equals(NO_ID)) {
+            // bridgeHandler.unregisterLightStatusListener(this);
+            stopAutomowerPolling();
+            automowerId.set(NO_ID);
+        }
+    }
+
+    private void startAutomowerPolling() {
+        if (automowerPollingJob == null || automowerPollingJob.isCancelled()) {
+            if (config.getPollingInterval() < 1) {
+                logger.info("No valid polling interval specified. Using default value: {}s", automowerPollingIntervalS);
+            } else {
+                automowerPollingIntervalS = config.getPollingInterval();
+            }
+            automowerPollingJob = scheduler.scheduleWithFixedDelay(new AutomowerPollingRunnable(), 1,
+                    automowerPollingIntervalS, TimeUnit.SECONDS);
+        }
+    }
+
+    private void stopAutomowerPolling() {
+        if (automowerPollingJob != null && !automowerPollingJob.isCancelled()) {
+            automowerPollingJob.cancel(true);
+            automowerPollingJob = null;
+        }
+    }
+
+    private boolean isValidResult(Mower mower) {
+        return mower != null && mower.getAttributes() != null && mower.getAttributes().getMetadata() != null
+                && mower.getAttributes().getBattery() != null && mower.getAttributes().getSystem() != null;
+    }
+
+    private boolean isConnected(Mower mower) {
+        if (mower != null && mower.getAttributes() != null && mower.getAttributes().getMetadata() != null
+                && mower.getAttributes().getMetadata().isConnected()) {
+            return true;
+        }
+        return false;
```

Comment:
`mower` cannot be null at this point. You could remove the null check.
```suggestion
        return mower.getAttributes() != null && mower.getAttributes().getMetadata() != null
                && mower.getAttributes().getMetadata().isConnected();
```

Reply:
thx